archethic/contracts/token_pool.exs

```diff
@@ -105,28 +122,38 @@ end
 # Archethic => EVM : Request secret hash #
 ##########################################
 
-condition triggered_by: transaction, on: request_secret_hash(htlc_genesis_address, amount, user_address, chain_id), as: [
-  type: "transfer",
-  code: valid_signed_code?(htlc_genesis_address, amount, user_address),
-  content: List.in?([@CHAIN_IDS], chain_id),
-  token_transfers:
-    (
-      valid? = false
-
-      htlc_genesis_address = String.to_hex(htlc_genesis_address)
-      transfers = Map.get(transaction.token_transfers, htlc_genesis_address, [])
-
-      for transfer in transfers do
-        if transfer.token_address == @TOKEN_ADDRESS &&
-             transfer.token_id == 0 &&
-             transfer.amount == amount do
-          valid? = true
-        end
-      end
+condition triggered_by: transaction, on: request_secret_hash(htlc_genesis_address, amount, user_address, chain_id) do
+  if transaction.type != "transfer" do
+    throw message: "Invalid transaction's type", code: 400
+  end
 
-      valid?
-    )
-]
+  if !valid_signed_code?(htlc_genesis_address, amount, user_address) do
+    throw message: "Invalid HTLC's code", code: 400
+  end
+
+  if !List.in?([@CHAIN_IDS], chain_id) do
+    throw message: "Invalid EVM chain id", code: 400
+  end
+
+  htlc_genesis_address = String.to_hex(htlc_genesis_address)
+  transfers = Map.get(transaction.token_transfers, htlc_genesis_address, [])
+
+  for transfer in transfers do
+    if transfer.token_address != @TOKEN_ADDRESS do
```

Comment:
This condition is a bit to strict I believe. It forbid user to make other transfer in their transaction while they could make transaction with multiple recipients or transfers if they wants